specs/core/0_fork-choice.md

```diff
@@ -63,17 +63,30 @@ class Target(object):
     root: Hash
 ```
 
+#### `RootSlot`
+
+```python
+@dataclass
+class RootSlot(object):
+    slot: Slot
```

Comment:
What is the reason for this field being a `Slot` (as opposed to an `Epoch`)? If we can go with `Epoch` then we could define a `Checkpoint` type to replace both `Target` and `RootSlot`.

Reply:
We can do that, and agreed it's cleaner.

In some places, we need to know the slot because of the the state lookup and the restriction of the consideration of blocks only prior to the 0th slot in the epoch (rather than just prior to the current epoch). In these cases, we'll do the `get_epoch_start_slot` conversion.

Reply:
Ah, the reason is that states are stored for various slots (some checkpoints, some just block states). Thinking about how to clean this up properly..